core/primitives/src/offchain.rs

```diff
@@ -19,6 +19,37 @@
 use rstd::prelude::{Vec, Box};
 use rstd::convert::TryFrom;
 
+/// A type of supported crypto.
+#[derive(Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+#[repr(C)]
+pub enum StorageKind {
+	///	Persistent storage is non-revertible and not fork-aware. It means that any value
+	/// set by the offchain worker triggered at block `N(hash1)` is persisted even
+	/// if that block is reverted as non-canonical and is available for the worker
+	/// that is re-run at block `N(hash2)`.
+	/// This storage can be used by offchain workers to handle forks
+	/// and coordinate offchain workers running on different forks.
+	PERSISTENT = 1,
+	/// Local storage is revertible and fork-aware. It means that any value
+	///	set by the offchain worker triggered at block `N(hash1)` is reverted
+	/// if that block is reverted as non-canonical and is NOT available for the worker
+	/// that is re-run at block `N(hash2)`.
+	LOCAL = 2,
```

Comment:
For `LOCAL` mode, will there be a way to query previous value (query at block N - x) ? (and a pruning that could match existing block pruning)

Reply:
Do you mean RPC or from within offchain workers?
1. RPC is not yet there, but most likely will be useful and should definitely allow querying at some particular (not-pruned block)
2. From within workers, I don't think so - they will only have access to the latest state.

Regarding pruning - yes, would be best to match current state pruning that is already in place.